private/pkg/storage/multi.go

```diff
@@ -76,6 +76,15 @@ func newMultiReadBucket(
 }
 
 func (m *multiReadBucket) Get(ctx context.Context, path string) (ReadObjectCloser, error) {
+	if m.overlay {
+		// If overlay is enabled, attempt a Get operation against the first bucket. If the file is
+		// found, we avoid the potentially expensive Stat call.
+		if readObjectCloser, err := m.delegates[0].Get(ctx, path); err != nil && !IsNotExist(err) {
+			return nil, err
+		} else if readObjectCloser != nil {
```

Comment:
Also add comment "if we did find a ReadObjectCloser, return it, otherwise do the expensive logic to check every delegate"

You could also optimize this further to skip the first delegate via a `bool` to m.getObjectInfoAndDelegateIndex